### PR TITLE
REF: move SparseDtype, ArrowDtype to dtypes.dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/util/__init__.py [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
-    Appender,
-    Substitution,
-    cache_readonly,
-)
-
-from pandas.core.util.hashing import (  # noqa: F401
-    hash_array,
-    hash_pandas_object,
-)
+def __getattr__(key: str):
```
Comment (mroeschke) [inline pandas/util/__init__.py:1]: Are these changes related to this PR?

**Comment by jbrockmendel** [comment]
Yes. Without this we raise at import time bc of circularity

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel
